src/classes/BDI_DataImportService.cls

```diff
@@ -72,6 +72,27 @@ global with sharing class BDI_DataImportService {
         return bdiMS;
     }
 
+    /*******************************************************************************************************
+    * @description constructs the soql String to run a Data Import process for list of DataImport ids.
+    * @param dataImportIds list of data import ids to include in query
+    * @return String the soql String
+    */
+    public static String strSoqlForDataImportProcess(List<Id> dataImportIds) {
+        String dataImportQuery;
+
+        // construct or SOQL for the batch
+        dataImportQuery = 'SELECT ' + String.join(listStrDataImportFields, ',');
```

Comment:
Is `listStrDataImportFields` always a full list of all of the fields on DI?  If so does this change mean that all fields will always be queried in both dry run and regular processing?

Reply:
So one thing I learned during this is that we are already querying all fields in regular processing - that's actually what the strSoqlForBatchProcess(Id batchId) method is handling. When I started building this I was trying to gather all the necessary fields for dry run more carefully, but what I learned is that for at least some objects (Address for example), dry run references every mapped field.